spec/plurimath/asciimath/parser_spec.rb

```diff
@@ -0,0 +1,505 @@
+require_relative "../../../lib/plurimath/math"
+require "strscan"
+
+RSpec.describe Plurimath::Asciimath::Parser do
+
+  it "returns instance of Cos formula against the string" do
+    asciimath = init_spec_resp("cos(2)")
+    # initializing common used values in respective named variables
+    cos = asciimath.value.first
+    cos_value = cos.angle.value
+    # cos formula
+    expect(cos.class).to eql(Plurimath::Math::Function::Cos)
+    expect(cos_value[0].value).to eq("(")
+    expect(cos_value[1].value).to eq("2")
+    expect(cos_value[2].value).to eq(")")
```

Comment:
Probably we don't need to store parentheses here, @ronaldtse right?

Reply:
Agree. I don't think a parenthesis is just a symbol. The point of a parenthesis is to wrap an expression. An expression can contain the parenthesis symbols as attributes, but the semantics are different.

Reply:
Here, cosine is a function that includes an expression. An expression could contain numbers. The current parse tree does not reflect this at all.